huaweicloud/services/mpc/resource_huaweicloud_mpc_transcoding_template_group.go

```diff
@@ -186,224 +187,325 @@ func ResourceTranscodingTemplateGroup() *schema.Resource {
+		d.Set("audio", flattenTemplateGroupAudio(utils.PathSearch("audio", respBody, make([]interface{}, 0)))),
+		d.Set("video_common", flattenTemplateGroupVideoCommon(utils.PathSearch("video_common", respBody, make([]interface{}, 0)))),
+		d.Set("template_ids", flattenTemplateIds(utils.PathSearch("template_ids", respBody, make([]interface{}, 0)).([]interface{}))),
+	)
 
-	if len(templateGroupList) == 0 {
-		log.Printf("unable to retrieve MPC transcoding template group: %s", d.Id())
-		d.SetId("")
-		return nil
+	rawCommon := utils.PathSearch("common", respBody, make([]interface{}, 0))
+	if rawCommon != nil {
+		mErr = multierror.Append(
+			d.Set("low_bitrate_hd", utils.PathSearch("common.PVC", respBody, nil)),
+			d.Set("hls_segment_duration", utils.PathSearch("common.hls_interval", respBody, nil)),
+			d.Set("dash_segment_duration", utils.PathSearch("common.dash_interval", respBody, nil)),
+			d.Set("output_format", utils.PathSearch("common.pack_type", respBody, nil)),
+		)
+	}
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func GetTranscodingTemplateGroup(client *golangsdk.ServiceClient, groupId []string) (interface{}, error) {
+	httpUrl := "v1/{project_id}/template_group/transcodings"
+	getPath := client.Endpoint + httpUrl
+	getPath = strings.ReplaceAll(getPath, "{project_id}", client.ProjectID)
+	getPath = fmt.Sprintf("%s?group_id=%s", getPath, groupId[0])
+	getOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
 	}
-	templateGroup := templateGroupList[0]
 
-	templateIds := make([]string, len(*templateGroup.TemplateIds))
-	for i, v := range *templateGroup.TemplateIds {
-		templateIds[i] = strconv.FormatInt(int64(v), 10)
+	resp, err := client.Request("GET", getPath, &getOpt)
+	if err != nil {
+		return nil, err
 	}
 
-	mErr := multierror.Append(nil,
-		d.Set("region", config.GetRegion(d)),
-		d.Set("name", templateGroup.Name),
-		d.Set("audio", flattenAudio(templateGroup.Audio)),
-		d.Set("video_common", flattenVideoCommon(templateGroup.VideoCommon)),
-		d.Set("videos", flattenVideos(templateGroup.Videos)),
-		d.Set("template_ids", templateIds),
-		setCommonAttrs(d, templateGroup.Common),
-	)
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return nil, err
+	}
 
-	if err = mErr.ErrorOrNil(); err != nil {
-		return diag.Errorf("error setting MPC transcoding template group fields: %s", err)
+	templateGgroup := utils.PathSearch("template_group_list|[0]", respBody, nil)
+	if templateGgroup == nil {
+		return nil, golangsdk.ErrDefault404{}
 	}
 
-	return nil
+	return templateGgroup, nil
 }
 
-func resourceTranscodingTemplateGroupUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	config := meta.(*config.Config)
-	client, err := config.HcMpcV1Client(config.GetRegion(d))
-	if err != nil {
-		return diag.Errorf("error creating MPC client : %s", err)
+func flattenTemplateGroupVideos(resp interface{}) []map[string]interface{} {
+	rawArray := resp.([]interface{})
```

Comment:
Suggest type conversion in the flatten method and add success flag judgment.
```suggestion
	rawArray, ok := resp.([]interface{})
```
Same to other parts.